prelude/toolchains/cxx.bzl

```diff
@@ -41,12 +41,12 @@ def _system_cxx_toolchain_impl(ctx):
             linker_info = LinkerInfo(
                 linker = RunInfo(args = ["clang++"]),
                 linker_flags = ["-fuse-ld=lld"],
-                archiver = RunInfo(args = ["ar", "rcs"]),
+                archiver = RunInfo(args = ["ar"]),
```

Comment:
Why is this necessary?

Reply:
Defining `"rcs"` in the `archiver`'s `RunInfo` breaks with Conan when building a package from source. The problem is that we configure `ar rcs` as the archiver programm and the build system of the Conan package again passes similar flags, e.g. `cs`, to the archiver, resulting in a command line of the form `ar rcs cs ...`. This fails with an error of the form:

```
ar: libz.a: No such file or directory
```
As explained in https://github.com/madler/zlib/issues/588#issuecomment-992694506 .

If we set `ar` alone as the archiver then this problem goes away. However, then we need to make sure that Buck2's archive actions pass the correct arguments to `ar`, which is why this sets `use_archiver_flags = True`, see [here](https://github.com/facebookincubator/buck2/blob/6ea6a5e1cc836fc057e5d28e24d1c89a229259b8/prelude/cxx/archive.bzl#L17-L18).

Perhaps the core issue is that the `archiver` field conflates the archiver binary and the flags to be passed to the archiver in an action to create a static archive. If instead it followed the pattern of e.g. `CCompilerInfo` and there were separate `archiver` and `archiver_flags` fields then we could extract the archiver even in a configuration that explicitly configures `archiver_flags` and sets `use_archiver_flags = False`.

See also related [commit message](https://github.com/facebookincubator/buck2/pull/58/commits/5f04d41640730baedaeed98cbb247da567bba389).